opengovsg/plumber
# PLU-81: [IF-THEN-12] Implement step overrides / 1st step for If-Then

## Problem
For UX reasons, when editing an if-then step in a nested editor, we want to:
1. Show "If... then" instead of the usual "Action" / "Trigger"
2. Show the branch name instead of the app name
3. Disable changing the action (i.e. hide the "Choose app & event" substep)

## Solution
This PR implements the UX changes above.

**BEFORE**:
![Screenshot 2023-09-01 at 8 04 18 PM](https://github.com/opengovsg/plumber/assets/135598754/17ace00d-84fa-43fd-897a-67b66943638b)

**AFTER**:
![Screenshot 2023-09-01 at 8 02 11 PM](https://github.com/opengovsg/plumber/assets/135598754/d3d13d8f-c520-4a56-86bc-397b3c8c2bcd)


### IMPORTANT NOTE 1
There is an existing bug where the "conditions" multi-row field does not initially show the default blank row. This is a complex bug, and I address it in a later PR. For now, the workaround (when testing) is to click on "Test Step", then back to "Set up Action". This will force a re-render and make the default blank row show up.

### IMPORTANT NOTE 2
This PR does not implement hidden fields, so you will still see "FILE A BUG IF YOU SEE THIS" inside "Set up action"

## Tests
- Check that if-then steps display "If... Then" in the step header instead of "Action"  / "Trigger"
- Check that if-then steps display "the branch name" in the step header instead of the app name
- Check that if-then steps display do not have the "Choose app & event" substep.
- Check that if-then steps don't have a delete button in the header
- Regression test: check that other actions are not overridden, and display normally.
